File: src/main/java/pl/dmichalski/algorithms/_6_max_subarray_sum/MaxSubarraySumService1.java
package pl.dmichalski.algorithms._6_max_subarray_sum;

class MaxSubarraySumService1 {

    /**
     * O(n2) complexity
     */
    Integer maxSubarraySum(int[] values, int numbersCountToCountSum) {
        if (values == null || numbersCountToCountSum > values.length) {
            return null;
        }

        int max = Integer.MIN_VALUE;
        for (int i = 0; i < values.length - numbersCountToCountSum + 1; i++) {
            int tmpMax = 0;
            for (int j = 0; j < numbersCountToCountSum; j++) {
                tmpMax += values[i + j];
            }
            if (tmpMax > max) {
                max = tmpMax;
            }
        }

        return max;
    }
}

File: src/main/java/pl/dmichalski/algorithms/_6_max_subarray_sum/MaxSubarraySumService1.kt
package pl.dmichalski.algorithms._6_max_subarray_sum

internal class MaxSubarraySumService1 {

    /**
     * O(n2) complexity
     */
    fun maxSubarraySum(values: IntArray?, numbersCountToCountSum: Int): Int? {
        if (values == null || numbersCountToCountSum > values.size) {
            return null
        }

        var max = Int.MIN_VALUE
        for (i in 0 until values.size - numbersCountToCountSum + 1) {
            var tmpMax = 0
            for (j in 0 until numbersCountToCountSum) {
                tmpMax += values[i + j]
            }
            if (tmpMax > max) {
                max = tmpMax
            }
        }

        return max
    }

}
